api/v1beta1/awscluster_webhook.go

```diff
@@ -115,6 +115,17 @@ func (r *AWSCluster) ValidateUpdate(old runtime.Object) error {
 					r.Spec.ControlPlaneLoadBalancer.Name, "field is immutable"),
 			)
 		}
+
+		// Block the update for HealthCheckProtocol :
+		// - if it was not set in old spec but added in new spec
+		// - if it was set in old spec but changed in new spec
+		if newLoadBalancer.HealthCheckProtocol != nil && existingLoadBalancer.HealthCheckProtocol == nil && newLoadBalancer.HealthCheckProtocol.String() != ClassicELBProtocolSSL.String() ||
```

Comment:
Thinking: could not we check this using `reflect.DeepEqual(newLoadBalancer.HealthCheckProtocol , existingLoadBalancer.HealthCheckProtocol)`

Reply:
I think it makes sense to use `reflect.DeepEqual()`, thanks @shivi28.